cas-server-documentation/integration/Delegate-Authentication.md

```diff
@@ -287,7 +245,22 @@ To start authentication on a remote provider, these links must be added on the l
 </form>
 {% endhighlight %}
 
+
 ##Demo
 
 Take a look at this demo: [cas-pac4j-oauth-demo](https://github.com/leleuj/cas-pac4j-oauth-demo) to see this authentication delegation mechanism in action.
 
+
+##Wrapper for pac4j authenticator and profile creator
+
+It's another feature: the `cas-server-support-pac4j` module has a wrapper to use a pac4j authenticator and profile creator as a CAS authentication handler.
+
+It has a first implementation for username/password credentials: `org.jasig.cas.support.pac4j.authentication.handler.support.UsernamePasswordWrapperAuthenticationHandler` (extended from `AbstractWrapperAuthenticationHandler`):
+
+{% highlight xml %}
+<bean id="simpleAuthenticator" class="org.pac4j.http.credentials.authenticator.test.SimpleTestUsernamePasswordAuthenticator" />
+
+<bean id="primaryAuthenticationHandler" class="org.jasig.cas.support.pac4j.authentication.handler.support.UsernamePasswordWrapperAuthenticationHandler">
```

Comment:
Not excatly, there is one root wrapper: `AbstractWrapperAuthenticationHandler` which is independent from the credentials type, because we must know how to convert the CAS credentials into pac4j credentials. Then, we have a first implementation: `UsernamePasswordWrapperAuthenticationHandler` for `UsernamePasswordCredential`.

Though, generic parameters are missing to ensure the right type of credentials (CAS & pac4j). Will add.